demos/imagenet_classification_demo/classification_results.h

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2019 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <iostream>
+#include <utility>
+
+#include <ie_blob.h>
+
+class ClassificationResult {
+public:
+    explicit ClassificationResult(InferenceEngine::Blob::Ptr output_blob,
+                                  size_t batch_size = 1,
+                                  size_t num_of_top = 10,
+                                  std::vector<std::string> image_names = {},
+                                  std::vector<std::string> labels = {}) :
+            _nTop(num_of_top),
+            _outBlob(std::move(output_blob)),
+            _labels(std::move(labels)),
+            _imageNames(std::move(image_names)),
+            _batchSize(batch_size) {
+    }
+
+    std::vector<unsigned> topResults(unsigned int n, InferenceEngine::Blob& input) {
+        std::vector<unsigned> output;
+
+        switch (input.getTensorDesc().getPrecision()) {
+            case InferenceEngine::Precision::FP32: {
+                using currentBlobType = InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP32>::value_type;
+                InferenceEngine::TBlob<currentBlobType>& tblob = dynamic_cast<
+                                                                    InferenceEngine::TBlob<currentBlobType>&>(input);
+                topResults(n, tblob, output);
+                break;
+            }
+            case InferenceEngine::Precision::FP16: {
+                using currentBlobType = InferenceEngine::PrecisionTrait<InferenceEngine::Precision::FP16>::value_type;
+                InferenceEngine::TBlob<currentBlobType>& tblob = dynamic_cast<
+                                                                    InferenceEngine::TBlob<currentBlobType>&>(input);
+                topResults(n, tblob, output);
+                break;
+            }
+            default:
+                THROW_IE_EXCEPTION << "cannot locate blob for precision: " << input.getTensorDesc().getPrecision();
+        }
+
+        return output;
+    }
+
+private:
+    size_t _nTop;
+    InferenceEngine::Blob::Ptr _outBlob;
+    const std::vector<std::string> _labels;
+    const std::vector<std::string> _imageNames;
+    const size_t _batchSize;
+
+    template<class T>
+    inline void topResults(unsigned int n, InferenceEngine::TBlob<T> &input, std::vector<unsigned> &output) {
```

Comment:
I think it is clearer to return output vector instead of filling the provided one, so the third argument should be removed. It might also be more convenient to return `vector<vector<unsigned>>` - batch of n best predictions.

Minor comment: the first and the second arguments should be swapped: `n` could have a default value and `input` couldn't.